lib/manager/jenkins/extract.ts

```diff
@@ -1,12 +1,69 @@
+import is from '@sindresorhus/is';
+import yaml from 'js-yaml';
 import * as datasourceJenkins from '../../datasource/jenkins-plugins';
 import { logger } from '../../logger';
 import { SkipReason } from '../../types';
 import { isSkipComment } from '../../util/ignore';
 import * as dockerVersioning from '../../versioning/docker';
 import type { PackageDependency, PackageFile } from '../types';
 
-export function extractPackageFile(content: string): PackageFile | null {
-  logger.trace('jenkins.extractPackageFile()');
+function getDependency(
+  name: string,
+  version: string,
+  url: string,
+  groupId: string
+): PackageDependency {
+  const dep: PackageDependency = {
+    datasource: datasourceJenkins.id,
+    versioning: dockerVersioning.id,
+    depName: name,
+  };
+
+  if (version) {
+    dep.currentValue = version.toString();
+  } else {
+    dep.skipReason = SkipReason.NoVersion;
+  }
+
+  if (version === 'latest' || version === 'experimental' || groupId) {
+    dep.skipReason = SkipReason.UnsupportedVersion;
+  }
+
+  if (url) {
+    dep.skipReason = SkipReason.InternalPackage;
+  }
+
+  logger.debug({ dep }, 'Jenkins plugin dependency');
+  return dep;
+}
+
+function extractYaml(content: string): PackageDependency[] {
+  const deps: PackageDependency[] = [];
+
+  try {
+    const doc = yaml.safeLoad(content, { json: true }) as any;
+    if (doc?.plugins && is.array(doc.plugins)) {
+      for (const plugin of doc.plugins) {
+        if (plugin.artifactId) {
+          // TODO: how can we specify if a dependency should be ignored with yaml ?
+          const dep = getDependency(
+            plugin.artifactId,
+            plugin.source?.version,
+            plugin.source?.url,
+            plugin.groupId
+          );
+          deps.push(dep);
+        }
+      }
+    }
+  } catch (err) /* istanbul ignore next */ {
```

Comment:
```suggestion
  } catch (err) {
```

Please add a simple test instead

Reply:
Added